### DISCUSSION TITLE
OTA Problems

### DISCUSSION THREAD

**Discussion by wa0uwh** [easytarget/esp32-cam-webserver#238] [Q&A] [answered]
I have been using esp32-cam-webserver V4.0 and the previous revisions for several months, but I still have not got the OTA to compile.  So far, I have had to add "#ifndif NO_OTA" around the OTA code section to avoid compile errors.

I compile on Arduino 1.8.19 IDE

But, now it is time for me to resolve my OTA problem, and could use some help.  I have updated the Boards, and updated the installed Lib's.   I have followed the OTA section in the GitHub README.  The only thing that I can think of is that I may have conflicting Lib's installed, but do not know which are needed and where others may conflict.  

The following are the first 3 errors:
```
C:\Users\eldonb\Dropbox\Cloud\A1Projects\Arduino\ESP32_CAM\esp32-cam-webserver-4.0\daschr\esp32-cam-webserver_MultiStream\esp32-cam-webserver_MultiStream.ino: In function 'void setup()':
esp32-cam-webserver_MultiStream:707:20: error: 'class ArduinoOTAMdnsClass<WiFiServer, WiFiClient, WiFiUDP>' has no member named 'setHostname'
         ArduinoOTA.setHostname(mdnsName);
                    ^~~~~~~~~~~
esp32-cam-webserver_MultiStream:710:24: error: 'class ArduinoOTAMdnsClass<WiFiServer, WiFiClient, WiFiUDP>' has no member named 'setPassword'
             ArduinoOTA.setPassword(otaPassword);
                        ^~~~~~~~~~~
esp32-cam-webserver_MultiStream:716:14: error: 'class ArduinoOTAMdnsClass<WiFiServer, WiFiClient, WiFiUDP>' has no member named 'onStart'
             .onStart([]() {
              ^~~~~~~
```

Any suggestion would be welcome.
-Eldonb

**Comment by easytarget** [comment, marked as answer]
This looks like your IDE + libraries are confused or corrupted; searching on the error messages will confirm this.
You will need to follow this up via the IDE help channels, I cannot debug it for you; the code _does_ compile correctly with the IDE+ libraries+boards configured as described in the README. 

Also, `esp32-cam-webserver_MultiStream.ino` is not what the sketch is called, eg you are on someone else's fork of the code, which is also something I cannot support.
